application/src/main/java/org/togetherjava/tjbot/commands/reminder/ReminderCommand.java

```diff
@@ -125,40 +159,84 @@ private void handleCreateCommand(SlashCommandInteractionEvent event) {
+            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
+            .fetch());
+    }
 
-        Result<PendingRemindersRecord> pendingReminders = getReminders(guildId, userId);
+    private MessageCreateData createListMessage(Result<PendingRemindersRecord> pendingReminders,
+            int pageToShow) {
+        int totalPages = Math.ceilDiv(pendingReminders.size(), REMINDERS_PER_PAGE);
+
+        EmbedBuilder remindersEmbed = new EmbedBuilder().setTitle("Pending reminders")
+            .setColor(RemindRoutine.AMBIENT_COLOR);
+        MessageCreateBuilder listMessage = new MessageCreateBuilder();
 
         if (pendingReminders.isEmpty()) {
             remindersEmbed.setDescription("No pending reminders");
         } else {
-            pendingReminders.forEach(reminder -> {
-                String content = reminder.getContent();
-                long channelId = reminder.getChannelId();
-                Instant remindAt = reminder.getRemindAt();
+            if (totalPages > 1) {
+                remindersEmbed.setFooter("Page: %d/%d".formatted(pageToShow, totalPages));
 
-                remindersEmbed.addField(content, getDescription.apply(channelId, remindAt), false);
-            });
+                getPageEntries(pendingReminders, pageToShow)
+                    .forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+
+                listMessage.addActionRow(createPageTurnButtons(pageToShow, totalPages));
+            } else {
+                pendingReminders.forEach(reminder -> addReminderAsField(reminder, remindersEmbed));
+            }
         }
 
-        event.replyEmbeds(remindersEmbed.build()).setEphemeral(true).queue();
+        return listMessage.addEmbeds(remindersEmbed.build()).build();
     }
 
-    private Result<PendingRemindersRecord> getReminders(long guildId, long userId) {
-        return database.read(context -> context.selectFrom(PENDING_REMINDERS)
-            .where(PENDING_REMINDERS.GUILD_ID.eq(guildId)
-                .and(PENDING_REMINDERS.AUTHOR_ID.eq(userId)))
-            .orderBy(PENDING_REMINDERS.REMIND_AT.asc())
-            .fetch());
+    private List<Button> createPageTurnButtons(int pageNumber, int totalPages) {
+        String pageNumberString = String.valueOf(pageNumber);
+
+        Button previousButton =
+                Button.primary(generateComponentId(pageNumberString), PREVIOUS_BUTTON_EMOJI);
+        if (pageNumber <= 1) {
+            previousButton = previousButton.asDisabled();
+        }
+
+        Button nextButton =
+                Button.primary(generateComponentId(pageNumberString), NEXT_BUTTON_EMOJI);
+        if (pageNumber >= totalPages) {
+            nextButton = nextButton.asDisabled();
+        }
+
+        return List.of(previousButton, nextButton);
+    }
+
+    private static List<PendingRemindersRecord> getPageEntries(
+            Result<PendingRemindersRecord> remindersRecords, int pageNumber) {
```

Comment:
maybe change to `paginateReminders`. and `pageNumber` should be changed to maybe `pageToDisplay` or `currentPage` or similar

Reply:
> maybe change to paginateReminders.

it doesn't paginate anything it just gives the reminders which should be shown on the given page

Reply:
yeah and `pageNumber` is ambigious. i didnt understand what it means when i read it. rename it so that its clear. maybe `pageToDisplay` or `pageToShowRemindersFor` or whatever.

Reply:
done